### ISSUE TITLE
Request to Open Source Fine-Tuning + Evaluation Code (Unsloth Version)

### ISSUE THREAD

**Issue by YangEgp** [JingbiaoMei/RGCL#3]
Hi, thank you for your amazing work on Qwen2-VL!

I’ve been trying to reproduce the fine-tuning (lora) results using Unsloth on Qwen2-VL-7B-Instruct, but the performance gap is huge — around 20% lower than the results reported in the paper on datasets like HARMP.

I’ve followed relevant GitHub issues and tried to match all settings, but the gap remains.
To make debugging and reproduction easier, could you please open source the exact Unsloth-based fine-tuning and evaluation code used in your experiments?

A single notebook or script would be super helpful. Thanks!

**Comment by JingbiaoMei** [comment]
Hi Yang,

I have already released the code for the fine-tuning experiments with Qwen models. I have not tried unsloth, but it shouldn't make any difference in terms of performance.

**Comment by YangEgp** [comment]
<img width="1248" alt="Image" src="https://github.com/user-attachments/assets/f52732a4-ac82-4c10-bf4f-d0d0779f5bd4" />
Hi, thanks for sharing your code. I followed your implementation exactly, but I found that the performance is still around 20% lower than reported in the paper. Could you kindly clarify if there are any additional steps, or tricks that are not mentioned in the repo?

**Comment by JingbiaoMei** [comment]
Did you use the released dataset file on the Huggingface?

**Comment by JingbiaoMei** [comment]
> Hi, I referred to the code you released and modified the source code of qwen2.5-VL in Transformers, which is suitable for your operation in stage1, such as LM head and so on. But the effect is not very ideal, and the effect of SFT and lora is very poor. Are there any key steps I might be missing?

You do not need to modify the code for Transformers library. The code base should just work for Qwen2.5-vl. For starting, please try follow the released config files for the baseline SFT experiments to see if you can get similar performance. Common problems in reproduction could be found here: https://github.com/JingbiaoMei/RGCL/issues/2

**Comment by NJIT-Fan** [comment]
> Did you use the released dataset file on the Huggingface?

Yes. I use the dataset file on the Huggingface. It has almost the same result.

<img width="736" alt="Image" src="https://github.com/user-attachments/assets/9e1d29b5-a49b-416c-b314-0d4ff29eb721" />

**Comment by YangEgp** [comment]
Hi, thank you for releasing the code and dataset!

While reproducing the training setup, I noticed that the released training set contains **228 samples that also appear in the test set**. This suggests potential data leakage, which could impact the validity of the reported performance.

Could you please clarify:
- Are these overlapping samples intended?
- If not, could this affect the evaluation results?

```
Common IDs (228):
memes_1103
memes_1129
memes_1156
memes_1187
memes_1189
memes_1232
memes_1269
memes_1273
memes_1283
memes_1298
memes_1305
memes_1324
memes_1340
memes_1343
memes_1347
memes_1353
memes_1403
memes_1412
memes_1414
memes_1424
memes_1428
memes_1533
memes_1551
memes_1571
memes_1575
memes_1581
memes_1588
memes_1598
memes_1659
memes_1689
memes_1690
memes_1701
memes_1702
memes_1715
memes_1716
memes_1766
memes_1986
memes_1996
memes_1997
memes_2002
memes_2031
memes_209
memes_2104
memes_2170
memes_2185
memes_2201
memes_221
memes_2243
memes_227
memes_2270
memes_2291
memes_230
memes_241
memes_253
memes_2679
memes_2707
memes_2730
memes_2775
memes_2779
memes_2780
memes_2815
memes_2833
memes_2836
memes_2838
memes_2867
memes_315
memes_3301
memes_3314
memes_3334
memes_3364
memes_3393
memes_3409
memes_3422
memes_3476
memes_3484
memes_350
memes_3590
memes_3605
memes_3608
memes_3618
memes_3634
memes_3640
memes_3641
memes_3696
memes_3703
memes_3730
memes_3754
memes_3759
memes_3761
memes_3770
memes_3772
memes_3773
memes_3785
memes_3823
memes_3824
memes_3863
memes_3879
memes_4233
memes_4246
memes_4261
memes_4344
memes_4345
memes_4366
memes_4367
memes_4372
memes_4410
memes_4431
memes_4444
memes_4451
memes_4467
memes_4469
memes_4471
memes_4499
memes_4500
memes_4526
memes_4539
memes_4582
memes_4600
memes_4651
memes_4674
memes_4707
memes_4715
memes_4735
memes_4736
memes_4739
memes_4747
memes_4769
memes_4782
memes_4790
memes_4811
memes_4815
memes_4838
memes_4846
memes_4873
memes_4884
memes_4893
memes_4896
memes_4909
memes_4934
memes_4937
memes_4964
memes_4973
memes_4994
memes_5020
memes_5021
memes_5039
memes_5051
memes_5100
memes_5162
memes_5660
memes_5678
memes_5696
memes_5743
memes_5788
memes_5851
memes_5862
memes_5872
memes_5885
memes_5961
memes_5995
memes_6054
memes_6078
memes_6101
memes_6193
memes_6299
memes_6302
memes_6335
memes_6349
memes_6355
memes_6459
memes_6495
memes_6539
memes_6544
memes_6553
memes_6562
memes_6563
memes_6568
memes_6608
memes_6631
memes_6638
memes_6654
memes_6661
memes_6662
memes_6713
memes_6722
memes_6723
memes_6725
memes_6732
memes_6752
memes_6754
memes_6789
memes_6803
memes_6849
memes_6852
memes_6860
memes_6890
memes_6902
memes_8061
memes_8065
memes_8067
memes_8096
memes_8121
memes_8129
memes_8135
memes_8155
memes_816
memes_8188
memes_8209
memes_8229
memes_8231
memes_8233
memes_8249
memes_8251
memes_827
memes_8271
memes_8277
memes_8290
memes_831
memes_834
memes_855
memes_872
memes_875
memes_891
memes_900
memes_93
memes_952
memes_960
memes_990
```

Thanks again for your great work!

**Comment by JingbiaoMei** [comment]
Hi Yang,

This is a known issue for HarmP-v0. There is also many duplicated examples in test splits and train splits. They have also released the Harm-P v1 that fixed this. However, since most of the previous work that we compare with uses the original v0, we also follows the same data setup to make sure the reported results are comparable with previous literature.

**Comment by YangEgp** [comment]
Thank you for your clarification! Are there similar situations(issues) in other datasets?

**Comment by JingbiaoMei** [comment]
I haven't encounter similar issues in other datasets.

**Comment by YangEgp** [comment]
Thanks a lot!
